packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -135,25 +148,27 @@ export class AwsS3Publish implements PublisherBase {
       return await new Promise<string>((resolve, reject) => {
         const entityRootDir = `${entityName.namespace}/${entityName.kind}/${entityName.name}`;
 
-        const fileStreamChunks: Array<any> = [];
         this.storageClient
           .getObject({
             Bucket: this.bucketName,
             Key: `${entityRootDir}/techdocs_metadata.json`,
           })
-          .createReadStream()
-          .on('error', err => {
+          .then(async file => {
+            const techdocsMetadataJson = await streamToString(
+              file.Body as Readable,
```

Comment:
yeah this cast is unfortunate, but barring any better solution right now, let's continue and then tweak some other time